docs/howtoguides/pro_in_airgapped.md

```diff
@@ -0,0 +1,102 @@
+2. `contracts-airgapped` runs the server for the Ubuntu Pro client
+3. `get-resource-tokens` fetches resource tokens for setting up mirrors of Canonical repositories (as we cannot access Canonical-hosted resources in an offline environment)
+
+
+## Get configuration to set up mirrors
+The Ubuntu Pro client cannot access Canonical-hosted services in an offline environment, so we will mirror these repositories locally.
+
+First, find out the repositories’ URLs of all the resources your contract is entitled to. Run the following in a non-airgapped environment: \
+`echo '[YOUR_CONTRACT_TOKEN]:' | ua-airgapped | grep 'aptURL:'`
+
+Then, get the resource tokens to access these resources in a non-airgapped environment: \
+`get-resource-tokens [YOUR_CONTRACT_TOKEN]`
+
+Stdout from `get-resource-tokens` gives pairs of services and their resource tokens (one token per service). The repository URL is specified as `aptURL` from the first command and, together with the resource token, they are required to set up a mirror, be it `apt-mirror`, `aptly`, Landscape or another software.
+
+You can check the resource token's correctness. For example, let’s say we want to set a mirror for `esm-infra`. It has `aptURL: https://esm.ubuntu.com`. If the resource token is `[ESM_INFRA_RESOURCE_TOKEN]`, then the command on the non-airgapped machine will be:
+```bash
+/usr/lib/apt/apt-helper download-file https://bearer:[ESM_INFRA_RESOURCE_TOKEN]@esm.ubuntu.com/ubuntu/ /tmp/check-esm-resource-token.txt
+```
+Response must be `200`. `401 Unauthorized` shows that the resource token or the URL is wrong. Please note that the specifics of setting up mirrors may be different depending on the product, but the bearer authentication will be identical in all cases, so refer to the command above to distinguish between airgapped and mirror problems.
+
+## Set up mirrors
+If you have experience in setting up mirrors, this section is optional for you. However, if you want to get started with the simplest option of setting up mirrors and have no prior experience, keep reading.
+
+Suppose you want to set up mirrors for esm-infra. Then, the steps to set up a mirror would be:
+1. Install `apt-mirror`: \
+`sudo apt install apt-mirror`
+2. Fetch the resource token for `esm-infra` (as per the section above).
+3. Change the `/etc/apt/mirror.list` file to the following contents:
+	```
+	set nthreads     20
+	set _tilde 0
+
+	deb https://bearer:[ESM_INFRA_RESOURCE_TOKEN]@esm.ubuntu.com/infra/ubuntu/ jammy-infra-updates main
+	deb https://bearer:[ESM_INFRA_RESOURCE_TOKEN]@esm.ubuntu.com/infra/ubuntu/ jammy-infra-security main
+
+	clean http://archive.ubuntu.com/ubuntu
+	```
+4. Run `apt-mirror`: \
+`sudo -u apt-mirror apt-mirror`
+5. Serve the `esm-infra` mirror on port `9090` (will be `[ESM_INFRA_MIRROR_URL]` in the next section): \
+`nohup sh -c 'python3 -m http.server --directory /var/spool/apt-mirror/mirror/esm.ubuntu.com/infra/ 9090 > /dev/null 2>&1 &'`
```

Comment:
I think we can change this command to just:
`python3 -m http.server --directory /var/spool/apt-mirror/mirror/esm.ubuntu.com/infra/ 9090`

Reply:
Done.